File: leetcode/easy/1365/1365.rb
# Runtime: 72 ms, faster than 74.00%
# Memory Usage: 210.5 MB, less than 45.00%
def smaller_numbers_than_current(nums)
  temp = nums.sort;
  table = [0]*101;

  cnt = 1
  1.upto(temp.size-1) do |i|
    table[temp[i]] = ((temp[i-1] == temp[i]) ? table[temp[i-1]] : cnt)
    cnt += 1
  end

  0.upto(nums.size - 1) do |i|
    temp[i] = table[nums[i]]
  end

  temp
end

File: leetcode/easy/1365/1365.js
/**
 * Runtime: 96 ms, faster than 68.18%
 * Memory Usage: 41 MB, less than 13.65%
 */
var smallerNumbersThanCurrent = function(nums) {
  let temp = [...nums];
  temp = temp.sort((a, b) => {return a-b;});
  let table = new Array(101).fill(0);
  let cnt = 1;
  table[nums[0]] = 0;
  for(let i=1; i<nums.length; ++i) {
    table[temp[i]] = ((temp[i-1] == temp[i]) ? table[temp[i-1]] : cnt);
    ++cnt;
  }

  for(let i=0; i<nums.length; ++i) {
    temp[i] = table[nums[i]];
  }

  return temp;
};
